server/src/main/java/org/elasticsearch/rest/RestController.java

```diff
@@ -827,7 +827,13 @@ public void sendResponse(RestResponse response) {
                     methodHandlers.addResponseStats(response.content().length());
                 } else {
                     final var wrapped = new EncodedLengthTrackingChunkedRestResponseBody(response.chunkedContent(), methodHandlers);
+                    final var headers = response.getHeaders();
                     response = RestResponse.chunked(response.status(), wrapped, Releasables.wrap(wrapped, response));
+                    for (final var header : headers.entrySet()) {
+                        for (final var value : header.getValue()) {
+                            response.addHeader(header.getKey(), value);
+                        }
```

Comment:
I wonder whether it is worth to reuse the existing headers instead of copying them over.